### ISSUE TITLE
The number of live traces spikes on one ingester

### ISSUE THREAD

**Issue by benmathews** [grafana/tempo#3761]
**Describe the bug**
I have 10 ingesters running. Every afternoon, one of the ten has a massive increase in the number of live traces. Normal is about 250K live traces per ingester. I've set the limit to 3 million and the one ingester reaches that. 

![image](https://github.com/grafana/tempo/assets/446380/c4f9f578-4605-4e12-a4e5-684c2b89a3f8)

During the problem period, the failing ingester will use a massive amount of RAM. Baseline is 12GiB. It spikes up to 35GiB. (I imagine this is correlated to the live traces limit but I haven't verified.)

Other than the following warning about live traces, I see nothing in the logs. 
```
level=warn ts=2024-06-07T17:32:23.761289997Z caller=instance.go:49 msg="LIVE_TRACES_EXCEEDED: max live traces exceeded for tenant single-tenant: per-user traces limit (local: 3000000 global: 0 actual local: 3000000) exceeded"
```
Other log lines
```
tempo-ingester-7 ingester level=info ts=2024-06-07T17:32:32.381898206Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=a3fac58a-d603-4d0f-b9ba-cd95fa9b781d
tempo-ingester-7 ingester level=info ts=2024-06-07T17:34:18.711194136Z caller=flush.go:260 msg="block completed" userid=single-tenant blockID=a3fac58a-d603-4d0f-b9ba-cd95fa9b781d duration=1m46.329300256s
tempo-ingester-7 ingester level=info ts=2024-06-07T17:34:18.825452883Z caller=flush.go:309 msg="flushing block" userid=single-tenant block=a3fac58a-d603-4d0f-b9ba-cd95fa9b781d
tempo-ingester-7 ingester level=info ts=2024-06-07T17:34:43.760453267Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=7a995814-615f-4f96-b23f-25c558b90bfe
tempo-ingester-7 ingester level=info ts=2024-06-07T17:34:43.760453267Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=7a995814-615f-4f96-b23f-25c558b90bfe
tempo-ingester-7 ingester level=info ts=2024-06-07T17:34:43.760453267Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=7a995814-615f-4f96-b23f-25c558b90bfe
```

The rate at which blocks are completed is far slower than on the functioning ingesters. 
```
tempo-ingester-0 ingester level=info ts=2024-06-07T17:31:59.50610174Z caller=flush.go:260 msg="block completed" userid=single-tenant blockID=954e213e-cc1e-4911-bd83-f62aeac3fa60 duration=40.433433904s
tempo-ingester-0 ingester level=info ts=2024-06-07T17:31:59.547320837Z caller=flush.go:309 msg="flushing block" userid=single-tenant block=954e213e-cc1e-4911-bd83-f62aeac3fa60
tempo-ingester-0 ingester level=info ts=2024-06-07T17:32:38.948435232Z caller=flush.go:177 msg="head block cut. enqueueing flush op" userid=single-tenant block=34e39e83-e1a5-4131-bb4e-75562d2f1a52
tempo-ingester-0 ingester level=info ts=2024-06-07T17:32:46.204436978Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=34e39e83-e1a5-4131-bb4e-75562d2f1a52
tempo-ingester-0 ingester level=info ts=2024-06-07T17:33:27.345732228Z caller=flush.go:260 msg="block completed" userid=single-tenant blockID=34e39e83-e1a5-4131-bb4e-75562d2f1a52 duration=41.1412971s
tempo-ingester-0 ingester level=info ts=2024-06-07T17:33:27.388298087Z caller=flush.go:309 msg="flushing block" userid=single-tenant block=34e39e83-e1a5-4131-bb4e-75562d2f1a52
tempo-ingester-0 ingester level=info ts=2024-06-07T17:33:59.013662693Z caller=flush.go:177 msg="head block cut. enqueueing flush op" userid=single-tenant block=8bd1df9c-d4ca-42e8-bb15-35417d1886c0
tempo-ingester-0 ingester level=info ts=2024-06-07T17:34:00.78430742Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=8bd1df9c-d4ca-42e8-bb15-35417d1886c0
tempo-ingester-0 ingester level=info ts=2024-06-07T17:34:41.496236508Z caller=flush.go:260 msg="block completed" userid=single-tenant blockID=8bd1df9c-d4ca-42e8-bb15-35417d1886c0 duration=40.711932565s
tempo-ingester-0 ingester level=info ts=2024-06-07T17:34:41.535495192Z caller=flush.go:309 msg="flushing block" userid=single-tenant block=8bd1df9c-d4ca-42e8-bb15-35417d1886c0
tempo-ingester-0 ingester level=info ts=2024-06-07T17:35:19.015316915Z caller=flush.go:177 msg="head block cut. enqueueing flush op" userid=single-tenant block=453f5f38-8c7a-401c-b932-2e28627a9c24
tempo-ingester-0 ingester level=info ts=2024-06-07T17:35:24.057458828Z caller=flush.go:253 msg="completing block" userid=single-tenant blockID=453f5f38-8c7a-401c-b932-2e28627a9c24
```

**To Reproduce**
I don't know. The behavior seems to happen regularly but without a known trigger.

**Expected behavior**
Ingester pods should continue completing and flushing blocks.

**Environment:**
 - Infrastructure: Self hosted Kubernetes cluster v1.26.10
 - Deployment tool: Helm tempo-distributed-1.10.0

**Additional Context**
Helm values file
[values.txt](https://github.com/user-attachments/files/15743165/values.txt)

**Comment by joe-elliott** [comment]
This is definitely odd. Some Qs that might help:

Are you in a multi or single tenant environment?
Is the ingester that has issues always the same one? Perhaps investigate other processes on the node that may be consuming a lot of resources or simply try moving it.

**Comment by benmathews** [comment]
This is a single tenant environment. 

I've been scaling up so haven't gone long periods between config changes. For the first three days this happened (6/4 - 6/6) it was ingenster-9. I made a config change and restarted all the ingesters on the 6th. Since then it is ingester-7. So yes, I think it always the same ingester save restarting the processes. 

CPU, memory, and network all look healthy w/ no changes between the normal and abnormal period. There is some evidence that disk was while not filling up, constrained. I gave it a bunch more and restarted. Will monitor to see if this resolves the problem.
